src/AgreementStructs.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.13;
+
+enum PositionStatus {
+    Idle,
+    Terminated
+}
+
+/// @dev Agreement party position
+struct Position {
+    /// @dev Matches index of the party in the agreement
+    uint256 id;
+    /// @dev Amount of tokens corresponding to this position.
+    uint256 balance;
+    /// @dev Status of the position
+    PositionStatus status;
+}
+
+struct Agreement {
+    /// @dev Hash of the detailed terms of the agreement.
+    bytes32 termsHash;
+    /// @dev Required amount to join or merkle root of (address,amount).
+    uint256 criteria;
+    /// @dev Number of termination confirmations.
+    uint256 terminations;
```

Comment:
Between `end` and `finalize` I think `finalize` works best because `finalizations` here is way more semantic than `endings`